django/contrib/auth/__init__.py

```diff
@@ -102,7 +102,7 @@ def login(request, user):
             request.session.flush()
     else:
         request.session.cycle_key()
-    request.session[SESSION_KEY] = user.pk
+    request.session[SESSION_KEY] = user._meta.pk.get_prep_value(user.pk)
```

Comment:
I think `value_to_string()` might be closer to what we want -- then every time we access the value, we need to use  `user._meta.pk.to_python(request.session[SESSION_KEY])` to get the value back.